test/e2e/e2e_suite_test.go

```diff
@@ -102,13 +103,17 @@ var recordedManifests = make(map[types.NamespacedName]io.Writer)
 // manifestRecorder keeps track of applied manifests
 type manifestRecorder struct {
 	client.Client
+	Mutex sync.Mutex
```

Comment:
I think we don't need to export it, and it is better give the same name with its type.

```suggestion
	mu sync.Mutex
```